src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +19,60 @@ public static Injector getInjector() {
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(
+                    "Class " + clazz.getName() + " is not annotated with @Component"
+            );
+        }
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+
+                clazzImplementationInstance = createNewInstance(clazz);
```

Comment:
The `clazzImplementationInstance` is being created inside the loop for each field annotated with `@Inject`. This means that if a class has multiple fields with `@Inject`, the instance will be recreated multiple times, which is inefficient and incorrect. Consider moving the instance creation outside the loop, before iterating over the fields.